### PR TITLE
[CAZB-4500] Handle CSV payment history download attempt

### PR DESCRIPTION
**Pull request by musztardem**
## Motivation and Context

https://eaflood.atlassian.net/browse/CAZB-4500
## Checklist:

- [ ] It contains only changes required by issue (does not contain other PR)
- [ ] Includes link to an issue (if apply)
- [ ] I have added tests to cover my changes.

### CODE REVIEW

File: app/controllers/payment_history/payment_history_controller.rb [hunk 1]
```diff
@@ -81,6 +82,35 @@ def payment_history_download
...
+    end
+
+    # Performs an API call to check the validity of the download URL and redirects accordingly.
+    #
+    # ==== Path
+    #
+    #   :GET /payment-history-export?exportId=87252eee-e861-4619-891e-30045908286c
+    #
+    def handle_payment_history_download_attempt
+      export_status = PaymentHistory::ExportStatus.new(account_id: current_user.account_id,
```
Comment (likeuwill) [inline app/controllers/payment_history/payment_history_controller.rb:103]: just `service`

File: app/controllers/payment_history/payment_history_controller.rb [hunk 2]
```diff
@@ -81,6 +82,35 @@ def payment_history_download
...
+                                                       job_id: export_id)
+
+      if export_status.link_active? && export_status.recipient?(current_user)
```
Comment (likeuwill) [inline app/controllers/payment_history/payment_history_controller.rb:106]: Can u move the logic to service method and then just call `if service.link_active?`

File: config/routes.rb [hunk 3]
```diff
@@ -203,6 +203,8 @@
     get :initiate_payment_history_download
     get :payment_history_downloading
     get :payment_history_download
+    get :payment_history_link_expired
+    get 'payment-history-export', to: 'payment_history#handle_payment_history_download_attempt'
```
Comment (likeuwill) [inline config/routes.rb:207]: `payment_history_export`

File: app/models/aws_signed_url.rb [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+
+  # Fetches creation datetime from URL and transforms to Time object.
+  def creation_time
+    Time.parse(parsed_query_params['X-Amz-Date']).utc
+  end
+
+  # Returns an integer which holds information about link activity in seconds e.g. 3600
+  def expiry_time
+    parsed_query_params['X-Amz-Expires'].to_i
+  end
```
Comment (likeuwill) [inline app/models/aws_signed_url.rb:26]: can be private methods

File: app/models/payment_history/export_status.rb [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+
+    # Returns a string with AWS signed URL
+    def file_url
+      api_data['fileUrl']
+    end
+
+    # Returns a string with job status
+    def status
+      api_data['status']
+    end
```
Comment (likeuwill) [inline app/models/payment_history/export_status.rb:35]: do we use this method?

**Review by likeuwill — CHANGES_REQUESTED** [review]

File: config/routes.rb (hunk 3 above)
Comment (musztardem) [reply config/routes.rb:207]: in the requirements we have dashes in the URL:
* https://eaflood.atlassian.net/browse/CAZB-4500
* https://github.com/InformedSolutions/JAQU-CAZ-Accounts-API/pull/613/files#diff-642bc44e6b65b953f54f016e9a3475290b3b7bc070f21f3814a355a631995228R73
and

**Review by likeuwill — APPROVED** [review]

**Pull request merged**